### ISSUE TITLE
(On going) Review Process

### ISSUE THREAD

**Issue by andreievg** [openmsupply/application-manager-server#88]
An outline of review schema and process, for discussion and reference (to be turned into internal docs and broken down into an epic)

**Comment by andreievg** [comment]
**From Applicant Perspective**

1. Can see 'changes request' when applications are in the list (I think we need to know how many observations have been edited)
2. When navigating to existing application page
 2.a Changes Requested View
 2.b What observations on what questions
3. What happens when new response is added

**1 Changes Request in the list**

User has apply rights on template
Current status of the application is changes required

**2.a Changes Requested View**

User has apply rights on template
Current status of the application is changes required

**2.b What observations on what questions**

Ideally this would be done by permissions schema. On top of the permission schema:

When application status is changes required
Take latest review
review_reponse_joins with send_to_appication_bool = true will be the observations.

** 3. What happens when new response is added ***

We can either create responses when changes requested application is opened, or when question is answered. 
Since currently our application logic relies on creating responses before application is edited maybe we should go with the former option. 

Have to think of a scenario where multiple users are being able to edit the same application, in which case, the best approach would be that later (for both creating application and dealing with requested changes).

![Screen Shot 2020-10-28 at 12 11 29 PM](https://user-images.githubusercontent.com/26194949/97371701-bb764f80-1916-11eb-8700-b9c92d19e66c.png)

Question with 'update' are the ones where latest application_response value is the same as application_response value that is related to the latest observation. Edit is where that value is different

**Comment by andreievg** [comment]
## New Schema suggestion

I am not sure if we need by section assignment and by section comment, as per: https://github.com/openmsupply/application-manager-server/issues/52#issuecomment-717569667

> Was wondering, do we really need a way to comment per section ? Can a comment by review be sufficient ?

> If that's the case we can simplify schema by removing review_section (and review_section_assignment). Assigner still chooses to assign by section, but internally we just work with review_question_assignment

In that case (and in the case that we still use joins), document named: Review with Joins 28-10

![Review with Joins 28-10](https://user-images.githubusercontent.com/26194949/97374957-7229fe00-191e-11eb-81b8-92d59940977e.png)

Basically when review goes to next level, joins are created rather then responses, and responses are created if modifications are made

**Comment by andreievg** [comment]
## From Assigner Perspective

**Comment by andreievg** [comment]
## From Reviewer Perspective

(including re-review)

**Comment by andreievg** [comment]
## From Consolidator Perspective

(including levels)
